### DISCUSSION TITLE
Is the conformal meshing technique utilized in Meep for modeling the surface between PEC and vacuum?

### DISCUSSION THREAD

**Discussion by tmteryz** [NanoComp/meep#2546] [General]
According to the Meep documentation, subpixel smoothing is only applicable to frequency-independent, lossless dielectrics. 
Therefore, I am curious if there are any techniques in Meep to enhance the accuracy of PEC/vacuum surface simulation. 
Additionally, what is the permittivity of a grid that contains both PEC and vacuum?

**Comment by smartalecH** [comment]
> Additionally, what is the permittivity of a grid that contains both PEC and vacuum?

Meep of course will simulate the material as if it had a permittivity of $-\infty$. But when the user queries the simulation about permittivity/permeability at these points, it actually defaults to air:

https://github.com/NanoComp/meep/blob/eb8ff3aff135f28e5171e2417817b6bbd9c98e7f/src/meepgeom.cpp#L827:L836

So this will have ramifications when plotting etc.
